pkg/syncer/sync.go

```diff
@@ -88,3 +88,31 @@ func (s *Syncer) Sync(charts ...string) error {
 
 	return errors.Trace(errs)
 }
+
+// SyncPendingCharts syncs the charts not found in the target
+//
+// It uses topological sort to sync dependencies first.
+func (s *Syncer) SyncPendingCharts(names ...string) error {
+	if err := s.loadChartsIndex(names...); err != nil {
+		return errors.Trace(err)
+	}
+	charts, err := s.topologicalSortCharts()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	var errs error
+	for _, chart := range charts {
+		id := fmt.Sprintf("%s-%s", chart.Name, chart.Version)
+		// 1. Process tgz file
```

Comment:
Is this WIP yet?

Reply:
yes! Do you prefer to take this method off of the PR?